packager/packager.cc

```diff
@@ -0,0 +1,665 @@
+            vector<ast::ParsedFile> results;
+            u4 filesProcessed = 0;
+            ast::ParsedFile job;
+            for (auto result = fileq->try_pop(job); !result.done(); result = fileq->try_pop(job)) {
+                if (result.gotItem()) {
+                    filesProcessed++;
+                    if (job.file.data(gs).sourceType == core::File::Type::Normal) {
+                        // Check if file path is part of a package.
+                        // TODO(jvilk): Could use a prefix array to make this lookup more efficient.
+                        auto path = job.file.data(gs).path();
+                        bool packaged = false;
+                        core::Context ctx(gs, core::Symbols::root(), job.file);
+                        for (const auto &pkg : packages) {
+                            if (absl::StartsWith(path, pkg->packagePathPrefix)) {
+                                job = rewritePackagedFile(ctx, move(job), pkg->name.mangledName);
+                                packaged = true;
+                                break;
+                            }
+                        }
+                        if (!packaged) {
+                            // Don't transform, but raise an error.
+                            if (auto e = ctx.beginError(job.tree->loc, core::errors::Packager::UnpackagedFile)) {
+                                e.setHeader("File `{}` does not belong to a package; add a `__package.rb` file to one "
+                                            "of its parent directories",
+                                            ctx.file.data(gs).path());
+                            }
+                        }
+                    }
+                    results.emplace_back(move(job));
+                }
+            }
+            if (filesProcessed > 0) {
+                resultq->push(move(results), filesProcessed);
+            }
+        });
+        files.clear();
+
+        {
+            vector<ast::ParsedFile> threadResult;
+            for (auto result = resultq->wait_pop_timed(threadResult, WorkerPool::BLOCK_INTERVAL(), gs.tracer());
+                 !result.done();
+                 result = resultq->wait_pop_timed(threadResult, WorkerPool::BLOCK_INTERVAL(), gs.tracer())) {
+                if (result.gotItem()) {
+                    files.insert(files.end(), make_move_iterator(threadResult.begin()),
+                                 make_move_iterator(threadResult.end()));
+                }
+            }
+        }
+    }
+
+    fast_sort(files, [](const auto &a, const auto &b) -> bool { return a.file < b.file; });
+
+    return files;
+}
+
+vector<ast::ParsedFile> Packager::runIncremental(core::GlobalState &gs, vector<ast::ParsedFile> files) {
+    // Just run all packages w/ the changed files through Packager again. It should not define any new names.
+    // TODO(jvilk): This incremental pass retypechecks every package file in the project. It should instead only process
```

Comment:
```suggestion
    // TODO(jvilk): This incremental pass reprocesses every package file in the project. It should instead only process
```

I think this is what you mean right? It doesn't actually run infer.

Reply:
yup